File: src/data/Service.js
import { types } from "mobx-state-tree"

const Service = types.model({
  id: types.maybe(types.integer),
  hourly_rate: types.maybe(types.integer),
  service_type: types.maybe(types.integer),
  name: types.string,
  position: types.integer,
})

export default Service

File: models/service.rb
class Service < ApplicationRecord
  # TODO change the database field to use strings,
  # to conserve the database seeds.
  enum service_type: ["mahjong", "pool", "ktv"]

  has_many :orders

  def as_json(*args)
    {
      id: id,
      hourly_rate: hourly_rate,
      name: name,
      position: position,
      service: service_type,
    }
  end
end
